yt_dlp/extractor/twitter.py

```diff
@@ -681,6 +702,114 @@ def _real_extract(self, url):
         return info
 
 
+class TwitterSpacesIE(TwitterBaseIE):
+    IE_NAME = 'twitter:spaces'
+    _VALID_URL = TwitterBaseIE._BASE_REGEX + r'i/spaces/(?P<id>[0-9a-zA-Z]{13})'
+    _TWITTER_GRAPHQL = 'https://twitter.com/i/api/graphql/HPEisOmj1epUNLCWTYhUWw/'
+
+    _TESTS = [{
+        'url': 'https://twitter.com/i/spaces/1RDxlgyvNXzJL',
+        'info_dict': {
+            'id': '1RDxlgyvNXzJL',
+            'ext': 'mp4',
+            'title': 'King Carlo e la mossa Kansas City per fare il Grande Centro',
+            'description': 'Twitter Space partecipated by annarita digiorgio, Signor Ernesto, Raffaello Colosimo, Simone M. Sepe',
+            'uploader': r're:Lucio Di Gaetano.+?',
+            'uploader_id': 'luciodigaetano',
+            'live_status': 'was_live',
+            'timestamp': 1659911386113,
+        },
+    }, {
+        # TODO: FIX TEST
+        # Twitter Space not started yet
+        'url': 'https://twitter.com/i/spaces/1YqGoAeoEOAxv',
+        'info_dict': {
+            'id': '1YqGoAeoEOAxv',
+            'ext': 'mp4',
+        },
+        'expected_warnings': ['Twitter Space not started yet'],
```

Comment:
@pukkandan is it possible to do a test to verify a raise_no_format situation?

Reply:
'params': {'ignore_no_formats_error': True}

Reply:
no, that's not what I was looking for.

`ignore_no_formats_error` basically skips the `raise_no_formats` line and goes on executing the code.

I'm looking to something like `'expected_error'` to actually feed a url that I know generates an error and test that situation. I guess is not something possible to test, am I right?